### ISSUE TITLE
Poor if statement formatting with Python 2.4 decompilation

### ISSUE THREAD

**Issue by pythonengineer** [rocky/python-uncompyle6#178]
When decompiling a Python 2.4 module, something such as this:

```
if True:
    pass
elif False:
    pass
elif not True:
    pass
```

will be decompiled as:

```
if True:
    pass
else:
    if False:
        pass
    else:
        if not True:
            pass
```

**Comment by pythonengineer** [comment]
@rocky I definitely understand, but this can actually be very annoying if there are several else ifs and it ends up looking something like this:

```
                                                                                                                                      if True:
                                                                                                                                          pass
```

Unfortunately, I don't really understand how the uncompyle internals work, and while I could learn them, I already have a lot of work to do with other projects, but I wish I could attempt to debug it. I made this issue more so that you and other devs know that it exists to be looked into in the future.

**Comment by rocky** [comment]
Sure, I can understand how busy you must be. There's no rush on this, s stuff like this can wait until someone finds the urge to address it.

**Comment by pythonengineer** [comment]
@rocky Something I forgot to ask, I was using whatever the latest commit was in the master branch at the time to get that result. I didn't notice that there was a "Python 2.4" branch, I did attempt to use it later but I couldn't get it to work since it had problems with the requirements for whatever reason, would that branch have contained a fix for this?

**Comment by alexsieusahai** [comment]
I want to take a crack at this; what would I need to know beforehand for ds and algs to understand the project beyond ast?

**Comment by rocky** [comment]
@alexsieusahai I have outlined what I think you need to know in https://github.com/rocky/python-uncompyle6/wiki/Adding-a-tree-transformation-phase-to-uncompyle6 . 

That has a link to other information, and generally the wiki has some information regarding how `uncompyle6` works.

**Comment by rocky** [comment]
Thanks to the efforts of @x0ret  this is now addressed in e96498ea.

Work has also been started on a [tree transformation phase](https://github.com/rocky/python-decompile3/pull/15)  in the [decompyle3](https://github.com/rocky/python-decompile3) project to be able to organize the code better, make this more understandable (at least to me) and be able to support things like this better in the future.
